notesapi/v1/tests/test_views.py

```diff
@@ -508,6 +507,40 @@ def test_pagination_page_start(self, page, start, previous_page, next_page):
             start=start
         )
 
+    def test_delete_all_user_annotations(self, user_id=TEST_USER):
+        """
+        Verify that deleting all user annotations works
+        """
+        self._create_annotation(text=u'Comment with foo', tags=[u'bar'])
+        self._create_annotation(text=u'Another comment', tags=[u'foo'])
+        self._create_annotation(text=u'A longer comment with bar', tags=[u'foo'])
+        response = self._get_search_results()
+        self.assertEqual(response["total"], 3)
+
+        url = reverse('api:v1:annotations')
+        self.headers["user_id"] = user_id
+        # Delete all notes for User 1
+        response = self.client.delete(url, self.headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Verify notes are deleted for User 1
+        response = self._get_search_results()
+        self.assertEqual(response["total"], 0)
+
+        # Reattempt delete for User 1
+        response = self.client.delete(url, self.headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Delete for No User
+        del self.headers["user_id"]
+        response = self.client.delete(url, self.headers)
```

Comment:
I see what's happening now. Django's `client.delete` takes args in the order `(path, data='', content_type='application/octet-stream', secure=False, **extra)` so `self.headers` is being passed as data to the request, not as headers.

Reply:
Makes perfect sense. Will update.